addons/website_crm_partner_assign/controllers/main.py

```diff
@@ -185,19 +185,18 @@ class WebsiteCrmPartnerAssign(WebsitePartnerPage):
     def sitemap_partners(env, rule, qs):
```

Comment:
Behavior:
Suppose that we have a Silver partner that is located in Belgium (like Gemini Furniture after changing its country to Belgium): if I deactivate the Silver partner level, when going to the `/partners` page what should we expect as the normal behavior ? :thinking: 
1.  Gemini Funiture not appearing at all (because of the disabled level) even though it has a country, and this country not appearing either because there is no partner to show ? (=> this is the behavior of your PR)
2. Gemini not appearing in the results of the level but still being listed thanks to its country ?

When removing a country, a partner still appears when filtering on the level and when filtering on "All Countries", so maybe it should behave the same the other way around (so not showing its level but still being displayed when filtering on "All Categories" and on the country, so still having its country displayed) ? :slightly_smiling_face:

Reply:
@sobo-odoo Maybe before going further, we could use the advice of @odoo/rd-sm-customer on this whole PR. If the whole active field seemed to serve no purpose, maybe we should not touch it in stable and we could just fix the publishing stuff? :shrug: 

@odoo/rd-sm-customer What do you think? :pray:

Reply:
@loco-odoo @sobo-odoo Let's go ahead with what SOBO was explaining:

- Inactive grades should not be listed on the partners page
- Active grades that have no partners: keep the current behavior (which is that the grade does not appear, right?)
- Grades that are not shown on the partners page (because inactive or no partner)
  - should not be listed in the sitemap
  - make their route do the same thing, which was chosen to be a 404, right?
- Do not change the current behavior for which _partners_ are listed (which, as SOBO explained, would be keeping listing them all because of the "All countries" filter?)

= Fix the sitemap/route behavior for grades, include the change that inactive grades means that the *grade* should not be listed, but make sure that the behavior regarding what *partners* are displayed stays the same.

(of course, if @odoo/rd-sm-customer disagrees, speak now or never :pray:)

Reply:
@qsm-odoo , thanks for your answer! Your firsts points are crystal clear, however, concerning
> Do not change the current behavior for which partners are listed (which, as SOBO explained, would be keeping listing them all because of the "All countries" filter?)

I still have a question: I think that what @sobo-odoo suggested is that if a partner is located in a country (e.g Belgium) and is linked to a grade (e.g Sliver) that is deactivated, it should still appears on the `/partners` url (not showing its level but still being displayed when filtering on "All Categories" and on the country, so still having its country displayed).
However, in the current behavior, if a partner is not linked to a grade, it does not appear at all on the `/partners` url (independently of the fact that the partner is located in a country or not).
Should we not consider that a partner linked to a deactivated grade is that same behavior than a partner not linked to any grade (and so not showing it at all on the `/partners` url)?